code/modules/spells/spell.dm

```diff
@@ -549,3 +550,33 @@ GLOBAL_LIST_INIT(spells, typesof(/obj/effect/proc_holder/spell)) //needed for th
 	user.visible_message("<span class='warning'>A wreath of gentle light passes over [user]!</span>", "<span class='notice'>You wreath yourself in healing light!</span>")
 	user.adjustBruteLoss(-10)
 	user.adjustFireLoss(-10)
+
+
+//Passive spells: new addition, might be useful when someone wants to make some new spells like this
+/obj/effect/proc_holder/spell/passive
+	charge_max = 0
+	cooldown_min = 0
+	var/toggle = FALSE
+	var/on_message = ""
+	var/off_message = ""
+
+//Always skip charge
+/obj/effect/proc_holder/spell/passive/charge_check(skipcharge = 1,mob/user = usr)
+	return TRUE
+
+/obj/effect/proc_holder/spell/passive/choose_targets(mob/user = usr)
+	perform(null,user)
+
+/obj/effect/proc_holder/spell/passive/perform()
+	to_chat(world,"perform called")
+	if(!toggle)
+		toggle = TRUE
+		invocation(usr)
+		to_chat(usr, "<span class='warning'>[on_message]</span>")
+		return
+
+	toggle = FALSE
+	to_chat(usr, "<span class='warning'>[off_message]</span>")
```

Comment:
This is technically alright, but it looks strange to leave this out of the if condition and make the if condition return on success. Normally we put the other code in an `else` statement like this:
```DM
if(!toggle)
	toggle = TRUE
	invocation(usr)
	to_chat(usr, "<span class='warning'>[on_message]</span>")
else
	toggle = FALSE
	to_chat(usr, "<span class='warning'>[off_message]</span>")
```

This will run the code only if the expression (`!toggle`) fails. Your code already does this, but not in a conventional way.

Reply:
Done